### DISCUSSION TITLE
GitHub Contribution Display Issue

### DISCUSSION THREAD

**Discussion by prmayanksingh** [community/community#112561] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

I'm facing an issue where my commits aren't showing in the contribution section. I've checked push, email, and waited, but no luck. Can you help?
![git hub 18 commit but not showing contribution](https://github.com/community/community/assets/149322455/969d1a56-cc4f-420b-89fa-74f9744d8c15)
![github contribution not showing](https://github.com/community/community/assets/149322455/3a12f932-f5ae-47fc-b64a-25b381eb2286)

**Comment by abdalla-am** [comment]
Make sure that u are allowing contributions to appear in your profile settings

**Comment by prmayanksingh** [comment]
My repository is public so I should automatically show

**Reply by abdalla-am** [reply, marked as answer]
if so ,thats a common problem that faces many github users

**Comment by simiaolin** [comment]
I have the same problem. I pushed a few commits to one of my public repositories. I have made sure that I correctly config my user name and user email, but my contributions are counted and shown on my Calendar. Anyone has some idea regarding this?

**Comment by arthurauffray** [comment]
Double-check all your account settings:
![Screenshot 2024-04-05 at 10 37 18 PM](https://github.com/community/community/assets/51604173/6118c998-22ef-450a-b88d-acc46e9a786a)

**Reply by arthurauffray** [reply]
https://github.com/settings/profile

**Reply by simiaolin** [reply]
@arthurauffray Thanks a lot for your help!

**Comment by Chamindu77** [comment]
If your commits aren't showing up in the contribution section of your GitHub profile, there could be several reasons for this. Here are some common troubleshooting steps you can take:

- Check the Email Address
- Verify Commit Authorship
- Check Repository Ownership
- Verify Pushes
- Check Repository Visibility
- Review Commit History
- Cache and Cookies
- Refresh GitHub Page

If after trying these steps your contributions still don't appear, you might want to reach out to GitHub support for further assistance. They can help diagnose the issue and provide personalized guidance based on your account and repository settings.

**Reply by simiaolin** [reply]
@Chamindu77  Thank you! 
After double-checking the email address, my contribution is shown correctly now. 
I agree that the first step is to check whether your user name and email address are correctly configured if your project is a public one.
Simply run the command `git log` and you will find the user name and email address in the git history.

**Comment by simiaolin** [comment]
For those who have the same issue as me and would like to rewrite the user information in previous git commits you could refer to [https://stackoverflow.com/questions/2919878/git-rewrite-previous-commit-usernames-and-emails](https://stackoverflow.com/questions/2919878/git-rewrite-previous-commit-usernames-and-emails).

**Comment by AhmadRazaNewMan** [comment]
when i make i new repo and push the code it appear in github contribution and and only show there  contribution and when i want to push more code the reach the repo but it does not appear in the github contribution on the same and on other  day as well  what is wrong i configure everythig but it is not working
